### ISSUE TITLE
Bug(bun test): afterEach not run for test in describe

### ISSUE THREAD

**Issue by H01001000** [oven-sh/bun#2528]
### What version of Bun is running?

0.5.8

### What platform is your computer?

Microsoft Windows NT 10.0.19044.0 x64 (also on github action)

### What steps can reproduce the bug?

Run `bun test` for following test file
```js
import { afterEach, beforeEach, describe, it } from 'bun:test'

describe('bucket.ts', () => {
  beforeEach(() => {
    console.log('before')
  })

  afterEach(() => {
    console.log('after')
  })

  describe('this is a test', () => {
    it('test a', () => {})
    it('test b', () => {})
  })
})
```

### What is the expected behavior?
(if the test where outside of describe('this is a test) it work)

before
✓ bucket.ts > this is a test > test a
after
before
✓ bucket.ts > this is a test > test b
after

### What do you see instead?

before
✓ bucket.ts > this is a test > test a
before
✓ bucket.ts > this is a test > test b

### Additional information

This is the test suite we are working on
https://github.com/discordeno/discordeno/pull/2798

Which this bug don't let us use sinon timer as it won't clear the time before next test
https://github.com/discordeno/discordeno/actions/runs/4580089374/jobs/8088476428?pr=2798

**Comment by blackmann** [comment]
Duplicate of #2487

**Comment by blackmann** [comment]
Resolved in `0.6.1`
